### PR TITLE
[14.0][OU-IMP] account: Post statements with reconciled items

### PR DESCRIPTION
**Pull request by pedrobaeza**
If you have statements in new state (open), but you have already performed some reconciliations on v13, you need to post the statements to continue doing reconciliations, and for having a consistent state between the lines of such statement, so we call the post button for achieving that, as only changing the statement state won't post the remaining journal entries.

@Tecnativa TT40209

### CODE REVIEW

File: openupgrade_scripts/scripts/account/14.0.1.1/post-migration.py [hunk 1]
```diff
@@ -324,6 +326,16 @@ def post_statements_with_unreconciled_lines(env):
...
+    # New statements with reconciled items
+    openupgrade.logged_query(
+        env.cr,
+        """SELECT bst.id FROM account_bank_statement bst
+        JOIN account_bank_statement_line bstl ON bstl.statement_id = bst.id
+        WHERE bst.state = 'open' AND bstl.is_reconciled IS TRUE
+        """,
+    )
+    stmt_ids = list({x[0] for x in env.cr.fetchall()})
+    env["account.bank.statement"].browse(stmt_ids).button_post()
```
Comment (MiquelRForgeFlow) [inline openupgrade_scripts/scripts/account/14.0.1.1/post-migration.py:338]: why not do this by query?

Comment (pedrobaeza) [reply openupgrade_scripts/scripts/account/14.0.1.1/post-migration.py:338]: Already explained in the main comment. The remaining moves won't be posted.

**Review by MiquelRForgeFlow — APPROVED** [review]

File: openupgrade_scripts/scripts/account/14.0.1.1/post-migration.py [hunk 2]
```diff
@@ -324,6 +326,16 @@ def post_statements_with_unreconciled_lines(env):
             AND bstl.is_reconciled IS DISTINCT FROM TRUE
         """,
     )
+    # New statements with reconciled items
+    openupgrade.logged_query(
+        env.cr,
+        """SELECT bst.id FROM account_bank_statement bst
```
Comment (MiquelRForgeFlow) [inline openupgrade_scripts/scripts/account/14.0.1.1/post-migration.py:332]:
```suggestion
        """SELECT DISTINCT bst.id FROM account_bank_statement bst
```

Comment (pedrobaeza) [reply openupgrade_scripts/scripts/account/14.0.1.1/post-migration.py:332]: Not needed, as I deduplicate later doing a Python set.

Comment (MiquelRForgeFlow) [reply openupgrade_scripts/scripts/account/14.0.1.1/post-migration.py:332]: Oh, right

**Pull request merged**